### PR TITLE
MECH MELEE UPDATE PART 2: AWESOME WEAPONS EDITION

### PR DESCRIPTION
**Pull request by Mqiib**
Adds more cool new mech melee weapons!
Weapons are as follows:

Maul: high damage, throws people away, bonus wounding, bonks structures, slow
FLAMING CHAINSWORD: lower damage than standard sword but burns and ignites targets hit, no deflection bonus
Katana: faster and lower damage sword with no cleave, wounds more, slightly better AP
Spear: +1 range, high AP, penetrates mechs to hit the pilot, slower, can be used (poorly) as a hand-held poke stick in a pinch
Stun Baton: works like a stun baton but can't NOT harmbaton because you're being smacked in the chest by a mech, deals 80 instead of 70 stamina damage and shoves people away
Rapier: Stabs twice, good AP, faster, low damage per stab, syndicate-only (dark gygax), mech steps forward to attack

TO DO:
- [x] Finish rapier
- [x] Fix the rapier lmao
- [x] Test all the weapons
- [x] Add a can stab check thing holy shit this was a wild ride
- [x] Add research stuff
- [x] Make/find sprites for all the weapons
- [x] Test everything again
- [ ] Convince people this is a good idea
- [ ] Testmerge (?) and do some thunderdome balance testing

# Spriting

Chainsword, spear, and rapier:
image

Maul, katana, and baton:
image

# Wiki Documentation

See description above of each weapon but word it probably a little less goofy, it's 4 AM and I'm tired and will probably redo this section sorry wiki people (mostly mord) if you get mad at me for being lazy on an unfinished pr 

# Changelog

:cl:  
rscadd: Adds several new mech-mounted melee weapons!
tweak: Lots of tweaks and changes in the background code of melee weapons to make them a bit more modifiable
tweak: MOST melee weapon damage vs mechs is reduced for the purpose of extending cool mech fights
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
mmmm mech weapons that can be used by a person, pretty epic

**Comment by Mantlecrawler** [comment]
woo flaming chainsaw mech woo

**Comment by maxion12345** [comment]
stunspear???

**Comment by ChuChu42** [comment]
suggestion: make the stunbaton deal 100 stun damage as its exepctionally hard to melee with a mech anyways and also its supposedly a giant stunbaton, it should deal more stam damage

another suggestion, Stun sword, make the stun baton into a sword with swing

**Comment by githubuser4141** [comment]
Shouldn't the exosuit be able to 'tap' a target with the stun baton, doing zero damage? I mean, excavators today, in real life, (we're supposdly 500 years behind Nanotrasen tech) can pick up eggs without breaking them, why can't a robotic vehicle in the future?

**Comment by Mqiib** [comment]
> suggestion: make the stunbaton deal 100 stun damage as its exepctionally hard to melee with a mech anyways and also its supposedly a giant stunbaton, it should deal more stam damage
> 
> another suggestion, Stun sword, make the stun baton into a sword with swing

1: I'm not giving mechs a one-tap weapon to instantly down people
2: Stun sword would be cool, I'll think about it but balancing it to where it would be useful but not outright better than other options would be a pain

**Comment by Mqiib** [comment]
> Shouldn't the exosuit be able to 'tap' a target with the stun baton, doing zero damage? I mean, excavators today, in real life, (we're supposdly 500 years behind Nanotrasen tech) can pick up eggs without breaking them, why can't a robotic vehicle in the future?

Fair point, I'll work on making the precise attack just do stamina damage.

**Comment by JamieD1** [comment]
Fucking mechs

**Review by ToasterBiome — APPROVED** [review]
Sprites fine

File: code/game/mecha/equipment/weapons/melee_weapons.dm [hunk 1]
```diff
@@ -190,54 +248,16 @@
...
-					playsound(O,'sound/weapons/smash.ogg', 50)		//metallic bonk noise
-
-		if(istype(T, /turf/closed/wall))		//IT BREAKS WALLS TOO
-			var/turf/closed/wall/W = T
-			W.dismantle_wall()
-
-	var/turf/cleave_effect_loc = get_step(get_turf(src), SOUTHWEST)	//Big sprite needs to be centered properly
-	new cleave_effect(cleave_effect_loc, chassis.dir)
+/obj/item/mecha_parts/mecha_equipment/melee_weapon/sword/energy_axe/special_hit(A)
+	if(istype(A, /turf/closed/wall))		//IT BREAKS WALLS TOO
```
Comment (ynot01) [inline code/game/mecha/equipment/weapons/melee_weapons.dm:262]: this will break rwalls jsyk

**Review by ynot01 — APPROVED** [review]
code fine

File: code/game/mecha/equipment/weapons/melee_weapons.dm (hunk 1 above)
Comment (Mqiib) [reply code/game/mecha/equipment/weapons/melee_weapons.dm:262]: Yes I am aware. It's a giant energy axe mounted on a nukie-only heavy mech (mauler) and mechs can break rwalls with their fists ANYWAYS.

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -976,7 +976,7 @@
 
 /datum/techweb_node/mech_maul
 	id = "mech_maul"
-	display_name = "Exosuit Weapon (CX-22 \"Barbados\" heavy maul)"
+	display_name = "Exosuit Weapon (CX-22 \"Barbatos\" heavy maul)"
```
Comment (MajManatee) [inline code/modules/research/techweb/all_nodes.dm:988]: https://en.wikipedia.org/wiki/Barbados

**Pull request merged**